test/e2e/features_test.go

```diff
@@ -94,8 +95,8 @@ func TestWebhookUpdate(t *testing.T) {
 	require.NoError(t, err)
 	addons := []clusters.Addon{}
 	addons = append(addons, metallb.New())
-	if shouldLoadImages() {
-		if b, err := loadimage.NewBuilder().WithImage(controllerImageOverride); err == nil {
+	if testenv.ClusterLoadImages() == "true" {
+		if b, err := loadimage.NewBuilder().WithImage(testenv.ControllerImageTag()); err == nil {
```

Comment:
Maybe we add a check for `testenv.ControllerImageTag()` when we hit this branch and then a nice error that you cannot load an empty image?

Reply:
The builder errors under that condition, so modified it to fail on the error.

Previously this would proceed if you asked it to load images but didn't indicate which, with the expectation that you're then using the stock images and don't need to load them. IMO actually failing in that case is fine--you should remove the unsatisfiable load request or specify an image--but it is a minor behavior change to be aware of.